src/Data/Semigroup/Cancellative.hs

```diff
@@ -248,24 +250,19 @@ instance RightReductive All where
 
 -- Identity & Const instances
 
-instance Reductive a => Reductive (Identity a) where
-   Identity a </> Identity b = Identity <$> (a </> b)
+deriving instance Reductive a => Reductive (Identity a)
 instance Reductive a => Reductive (Const a x) where
    Const a </> Const b = Const <$> (a </> b)
```

Comment:
I suspect we could use the same approach for `Const`, but I've tried to keep this PR specific to `Identity`.

Reply:
Good choice, thanks.